src/glmfit.jl

```diff
@@ -731,3 +763,65 @@ function checky(y, d::Binomial)
     end
     return nothing
 end
+
+"""
+    nobs(obj::LinearModel)
+    nobs(obj::GLM)
+
+For linear and generalized linear models, returns the number of rows, or,
+when prior weights of type FrequencyWeights are specified, the sum of weights.
+"""
+nobs(obj::LinPredModel) = nobs(obj.rr)
+
+nobs(r::LmResp{V,W}) where {V,W} = oftype(sum(one(eltype(r.wts))), length(r.y))
+nobs(r::LmResp{V,W}) where {V,W<:FrequencyWeights} = r.wts.sum
+
+nobs(r::GlmResp{V,D,L,W}) where {V,D,L,W<:FrequencyWeights} = r.wts.sum
+nobs(r::GlmResp{V,D,L,W}) where {V,D,L,W} = oftype(sum(one(eltype(r.wts))), length(r.y))
+
+## To be reviewed!
+Base.sqrt(::UnitWeights{T}) where T = one(T)
+
+function residuals(r::GlmResp; weighted::Bool = false)
+    ## Note: this is necessary if we want to be able to evsaluate
+    ## unweighted residuls when the model is weighted. Otherwise, if we
+    ## agree that the residulas should follow the specification of the model
+    ## we could use the object devresid 
+    y, η, μ = r.y, r.eta, r.mu
+    dres = similar(μ)
+
+    @inbounds for i in eachindex(y, μ)
+        μi = μ[i] 
+        yi = y[i]
+        dres[i] = sqrt(max(0, devresid(r.d, yi, μi)))*sign(yi-μi)        
+    end
+
+    if weighted
+        dres .= dres.*sqrt.(r.wts)
+    end
+
+    return dres
+end
+
+mdisp(rr::GlmResp{T1, <: Union{Normal, Poisson, Binomial, Bernoulli, NegativeBinomial}, T2, T3}) where {T1, T2, T3} = one(1)
+mdisp(rr::GlmResp{T1, <: Union{Gamma, Geometric, InverseGaussian}, T2, T3}) where {T1, T2, T3} = sum(abs2, rr.wrkwt.*rr.wrkresid)/sum(rr.wrkwt)
```

Comment:
I mean that `residuals` does:
```
    d = variancestructure(m.rr)
    r = m.rr.wrkwt .* m.rr.wrkresid
```
So it could do:
```
    r = m.rr.wrkwt .* m.rr.wrkresid
    d = variancestructure(m.rr, r)
```
So that `variancestructure` can do `sum(abs2, r)` instead of recomputing `Base.Broadcast.broadcasted(*, rr.wrkwt, rr.wrkresid))`.